storage/cloud-client/bucket_lock_test.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+
+import google.api_core.exceptions
+from google.cloud import storage
+
+import pytest
+
+import bucket_lock
+
+BLOB_NAME = 'storage_snippets_test_sigil'
+BLOB_CONTENT = 'Hello, is it me you\'re looking for?'
+# Retention policy for 5 seconds
+RETENTION_POLICY = 5
+
+
+@pytest.fixture()
+def bucket():
+    """Creates a test bucket and deletes it upon completion."""
+    client = storage.Client()
+    bucket_name = 'bucket-lock-' + str(int(time.time()))
+    bucket = client.create_bucket(bucket_name)
+    yield bucket
+    bucket.delete(force=True)
+
+
+def test_retention_policy_no_lock(bucket):
+    bucket_lock.set_retention_policy(bucket.name, RETENTION_POLICY)
+    bucket.reload()
+
+    assert bucket.retention_period is RETENTION_POLICY
+    assert bucket.retention_policy_effective_time is not None
+    assert bucket.retention_policy_locked is None
+
+    blob = bucket.blob(BLOB_NAME)
+    blob.upload_from_string(BLOB_CONTENT)
+
+    assert blob.retention_expiration_time is not None
+
+    bucket_lock.remove_retention_policy(bucket.name)
+    bucket.reload()
+    assert bucket.retention_period is None
+
+    time.sleep(RETENTION_POLICY)
+
+
+def test_retention_policy_lock(bucket):
```

Comment:
I recommend using a different bucket when locking the policy. IIRC the tests are ran out of order which may could impact other tests which upload objects to this test bucket.

Reply:
The pyfixture creates a new bucket for each test :)

Reply:
oooo!! NVM 👍 #TIL.

Reply:
👍